packages/monaco/src/browser/monaco-editor-model.ts

```diff
@@ -364,7 +366,7 @@ export class MonacoEditorModel implements IResolvedTextEditorModel, TextEditorDo
     }
 
     save(options?: SaveOptions): Promise<void> {
-        return this.scheduleSave(TextDocumentSaveReason.Manual, undefined, undefined, options);
+        return this.scheduleSave(options?.saveReason ?? TextDocumentSaveReason.Manual, undefined, undefined, options);
```

Comment:
I'm not quite sure why we differentiate the different "save" operations by reason.